File: src/main/java/org/tvl/tvlooker/domain/model/dto/Item.java
package org.tvl.tvlooker.domain.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.tvl.tvlooker.domain.model.enums.TmdbType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.Set;

@Builder
@Getter
@AllArgsConstructor
public class Item {
    private final Long id;
    private final String title;
    private final String overview;
    private final LocalDate releaseDate;
    private final BigDecimal popularity;
    private final BigDecimal voteAverage;
    private final TmdbType tmdbType;
    private final Long tmdbId;
    private final Set<Genre> genres;
    private final Set<Director> directors;
    private final Set<ActorItem> actorsInItem;
}

File: frontend/src/types/item.ts
import type { UUID } from "crypto";

export type Genre = {
  id: number;
  tmdbId: number;
  name: string;
}

export type Actor = {
  id: number;
  tmdbId: number;
  name: string;
}

export type ActorItem = {
    id: number;
    actorId: number;
    actorName: string;
    characterName: string;
    billingOrder: number;
}

export type Director = {
  id: number;
  tmdbId: number;
  name: string;
}

export type Item = {
  id: number;
  title: string;
  overview: string;
  releaseDate: string;
  popularity: number;
  voteAverage: number;
  type: "MOVIE" | "TV";
  tmdbId: number;
  genres: Genre[];
  actors: ActorItem[];
  directors: Director[];
  posterUrl?: string;
  backdropUrl?: string;
}

export type RecommendationResponse = {
  userId: UUID;
  items: Item[];
  count: number;
};
